### ISSUE TITLE
Use Matchers instead of separate test

### ISSUE THREAD

**Issue by DronMDF** [DronMDF/2out#322]
I thing about matchers...
```
Test(
    Text,
    Matcher1,
    Matcher2,
    Matcher3
)?
```

**Comment by DronMDF** [comment]
```c++
make_shared<MatchTest>(
     "12345",
     make_shared<ContainMatch>("3"),
     make_shared<StartWithMatch>("12"),
     make_shared<RegexMatch>("1.*5")
)
```

this is three separate Test results.
All matches sholud be complete.
Need more examples.

**Comment by DronMDF** [comment]
New idea:
use builders (Mix in) for Tests class
```c++
class ConcreteTest final : public 2out::Test, private 2out::Matchers
{
}
```
Any Matchers method available in initialization list
```c++
make_shared<MatchTest>(
     "12345",
     eqial("3"),
     startsWith("12"),
     match("1.*5")
)
```
Each method produce concrete matcher.
This way help us to avoid use make_shared...

**Comment by 0pdd** [comment]
@DronMDF 2 puzzles [#350](https://github.com/DronMDF/2out/issues/350), [#351](https://github.com/DronMDF/2out/issues/351) are still not solved.

**Comment by 0pdd** [comment]
@DronMDF the puzzle [#350](https://github.com/DronMDF/2out/issues/350) is still not solved; solved: [#351](https://github.com/DronMDF/2out/issues/351).

**Comment by 0pdd** [comment]
@DronMDF the puzzle [#373](https://github.com/DronMDF/2out/issues/373) is still not solved; solved: [#350](https://github.com/DronMDF/2out/issues/350), [#351](https://github.com/DronMDF/2out/issues/351).

**Comment by 0pdd** [comment]
@DronMDF all 3 puzzles are solved here: [#350](https://github.com/DronMDF/2out/issues/350), [#351](https://github.com/DronMDF/2out/issues/351), [#373](https://github.com/DronMDF/2out/issues/373).
